### DISCUSSION TITLE
Add "First line of text"

### DISCUSSION THREAD

**Discussion by livedevlab** [coddingtonbear/obsidian-web#101] [Ideas]
handlebars.ts
```
Handlebars.registerHelper("firstLine", (value: string): string => {
  return value.split("\n")[0];
});
Handlebars.registerHelper("filenameFirstLine", (unsafe: string | undefined): string => {
  if (typeof unsafe === "string") {
    return unsafe.split("\n")[0].replace(/[/\\?%*:|"<>]/g, "");
  }
  return "";
});
```

options.tsx

```
<li>
    <code>&#123;&#123;firstLine FIELD&#125;&#125;</code>: First line of text; 
    for example: <code>&#123;&#123;firstLine page.selectedText&#125;&#125;</code>
</li>
<li>
    <code>&#123;&#123;filenameFirstLine FIELD&#125;&#125;</code>: First line of text 
    and strips any characters from <code>FIELD</code> that are not safe in a filename.
</li>
```
<code>&#123;&#123;firstLine FIELD&#125;&#125;</code>: First line of text;  for example: <code>&#123;&#123;firstLine page.selectedText&#125;&#125;</code>
<code>&#123;&#123;filenameFirstLine FIELD&#125;&#125;</code>: First line of text and strips any characters from <code>FIELD</code> that are not safe in a filename.

**Comment by coddingtonbear** [comment]
Could you elaborate on what your use case is for this?  If it's to get some kind of excerpt of your text, I'd imagine that splitting on `\n` wouldn't be what you want given that newlines in HTML content are only loosely connected with a "line" on the screen.  I might be up for accepting a PR adding a handlebars function that returns an excerpt based on character or word count, though, maybe?
